internal/pipeline/pkgbuilder.go

```diff
@@ -0,0 +1,32 @@
+package pipeline
+
+import (
+	"io/ioutil"
+)
+
+// creates a directory and writes a Kptfile
+func writePkg(path string) (*pkg, error) {
+	dir, err := ioutil.TempDir(path, "")
+	if err != nil {
+		return nil, err
+	}
+	err = ioutil.WriteFile(dir+"/Kptfile", []byte(`
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+`), 0600)
+	if err != nil {
+		return nil, err
+	}
+	p, err := newPkg(dir)
+	return p, err
+}
+
+func (p pkg) withPipeline() *pkg {
+	p.pipeline = &Pipeline{}
+	return &p
+}
+
+func (p pkg) withSources(sources []string) *pkg {
+	p.pipeline.Sources = sources
+	return &p
+}
```

Comment:
`Pipeline` and `sources` should be read from the pkg itself, not sure why do we need these methods. If it is required to create arbitrary pipeline/source for a pkg, then we should avoid that. It will make maintaining tests harder.

Reply:
I realized, we can make the `resourceSources` to take just path and not be defined at the `pkg` level. That will make it easier to test.

Reply:
updated the PR to add a helper that takes a path to make tests easier. We can talk about this if you have further comments or suggestions